x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -51,17 +51,19 @@ const getCurrentValueFromAggregations = (
 
 const getParsedFilterQuery: (
   filterQuery: string | undefined
-) => Record<string, any> = filterQuery => {
+) => Record<string, any> | Array<Record<string, any>> = filterQuery => {
   if (!filterQuery) return {};
   try {
     return JSON.parse(filterQuery).bool;
   } catch (e) {
-    return {
-      query_string: {
-        query: filterQuery,
-        analyze_wildcard: true,
+    return [
+      {
+        query_string: {
+          query: filterQuery,
```

Comment:
At this point isn't this a Kuery filter? We should be using `convertKueryToElasticSearchQuery` in `x-pack/plugins/infra/public/utils/kuery.ts` to convert it.

Reply:
@simianhacker We discussed this, but `convertKueryToElasticSearchQuery` relies on UI code. In another PR, I run `convertKueryToElasticSearchQuery` on all of the filter queries on the frontend.

Reply:
If you look at that function you will see that it internally it links to some frontend code. BUT the same underlying functions are available on in the server part of the `data` plugin.

Reply:
So would it be redundant to add `convertKueryToElasticSearchQuery` on the frontend if we add it to the backend? Or vice versa? I'm wondering if the backend should just expect one or the other and not have a case handler for both of them.

The reason this bug happened is because the frontend used to send converted ES queries (which worked) and then at some point changed to KQL (which, as it turned out, did not work, but failed silently). So maybe we should just pick one and throw an error if we receive the wrong one?

Reply:
It shouldn't be a problem to always do it on the frontend 🤷‍♀️from what I can tell. That change is a bit more intrusive though. We basically need to start storing two fields `text` and `query`. Query is what the backend would use and text is what the frontend would display.

Reply:
Also, we need to do it on the frontend for validation at least, so that might be another argument for doing it on the frontend.

Reply:
@simianhacker How about we merge this as-is just to get filter queries working again, and then remove backend KQL support after merging https://github.com/elastic/kibana/pull/64292? That way the backend will only expect a converted query and error out if it's some other arbitrary string?

Reply:
Doing it on the front-end is sufficient, that's what we are doing on Metrics Explorer. I actually prefer the API's to be unaware of "Source", "Kuery", or whatever thing requires some special service to convert it into something ES can understand.

Reply:
Going to merge this just so we don't have broken functionality on `master` and track removing KQL from the backend in #64832